### ISSUE TITLE
Make sure int keys are handled by redis integration

### ISSUE THREAD

**Issue by antonpirker** [getsentry/sentry-python#3130]
In this PR:
https://github.com/getsentry/sentry-python/pull/3073/files#diff-79a3788011179b24960642acf26b7cd5a7daaef510906fb683364309338546fcR47

there is probably a string.join() that lead to this error:
https://sentry.sentry.io/issues/5448730468/?project=1&query=release%3Abackend%40a13563880494efef0811369c3265d546508f2703&referrer=release-issue-stream

Make sure that `_get_safe_command` and `_get_safe_key` can handle integers.

**Comment by MartinBraquet** [comment]
This issue has even more coverage than just int keys. It fails for any SCAN call.
```python
import redis
import sentry_sdk

sentry_sdk.init(dsn=...)

client = redis.Redis()
cursor, keys = client.scan()
```
Will raise
```
Traceback (most recent call last):
  File "_sentry_error.py", line 9, in <module>
    cursor, keys = client.scan()
  File "redis/commands/core.py", line 3026, in scan
    return self.execute_command("SCAN", *pieces, **kwargs)
  File "sentry_sdk/integrations/redis/_sync_common.py", line 72, in sentry_patched_execute_command
    cache_properties = _compile_cache_span_properties(
  File "sentry_sdk/integrations/redis/modules/caches.py", line 33, in _compile_cache_span_properties
    key_as_string = _key_as_string(key)
  File "sentry_sdk/integrations/redis/utils.py", line 61, in _key_as_string
    key = ", ".join(_safe_decode(x) for x in key)
TypeError: sequence item 0: expected str instance, int found
```
Thanks for you work

**Comment by sentrivana** [comment]
Fix out now in [2.5.0](https://github.com/getsentry/sentry-python/releases/tag/2.5.0)
